src/main/java/com/revature/controllers/TripController.java

```diff
@@ -107,13 +129,74 @@
      */
     @ApiOperation(value = "Updates a trip by ID", tags = {"Trip"})
     @PutMapping
-    public ResponseEntity<Trip> updateTrip(@Valid @RequestBody Trip trip) {
+    public ResponseEntity<TripDTO> updateTrip(@Valid @RequestBody Trip trip) {
         Optional<Trip> existingTrip = tripService.getTripById(trip.getTripId());
 
-        if (existingTrip.isPresent()) return ResponseEntity.status(201).body(tripService.updateTrip(trip));
+        if (existingTrip.isPresent()) return ResponseEntity.status(201).body(new TripDTO(tripService.updateTrip(trip)));
 
         return ResponseEntity.badRequest().build();
     }
+    
+    /**
+     * HTTP POST method (/trips/{tripId}/rider)
+     * 
+     * @param tripId represents the id of the trip the person is joining
+     * @param rider represents the User joining the trip.
+     * @return The newly updated Trip object.
+     */
+    @ApiOperation(value = "Adds a user to a trip", tags = {"Rider", "Trip"})
+    @PostMapping("/{tripId}/rider")
+    public ResponseEntity<TripDTO> updateTripRider(@PathVariable("tripId") int tripId, @Valid @RequestBody UserDTO rider){
+    	Optional<Trip> existingTrip = tripService.getTripById(tripId);
+    	Optional<User> existingUser = userService.getUserById(rider.getUserId());
+    	
+    	if(existingTrip.isPresent() && existingUser.isPresent()) {
+    		Trip trip = existingTrip.get();
+    		List<User> riders = trip.getRiders();
+    		riders.add(existingUser.get());
+    		int numberOfSeats = trip.getAvailableSeats() - 1;
+    		if(numberOfSeats < 0)return ResponseEntity.status(HttpStatus.CONFLICT).build();
+    		trip.setAvailableSeats(numberOfSeats);
+    		tripService.updateTrip(trip);
+    		return ResponseEntity.accepted().body(new TripDTO(trip));
+    	}else if(!existingTrip.isPresent()) {
+    		return ResponseEntity.notFound().build();
+    	}else {
+    		return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
```

Comment:
Is unauthorized the correct response code to use?

Reply:
This else can only be reached if the user is invalid. When I had an if(!existingUser.isPresent()) it was mad for an "always true" conditional check.

Reply:
I might recommend removing this else block, and removing the if on the block above (line 162). If either of the objects is not present, then we can return a Not Found.